src/publish.js

```diff
@@ -114,28 +114,33 @@ have published it.\
       };
 
       return new Promise((resolve, _reject) => {
-        const requestTags = () => void request.get(requestSettings, (_error, response, tags) => {
-          tags ??= [];
-          if (response?.statusCode === 200) {
-            if (tags.find(elem => elem.name === tag) != null) {
-              resolve();
-              return;
+        const requestTags = () => {
+          request.get(
+            requestSettings,
+            (_error, response, tags) => {
+              tags ??= [];
+              if (response?.statusCode === 200) {
+                if (tags.some(t => t.name === tag)) {
+                  resolve();
+                  return;
+                }
+              }
+              if (--retryCount <= 0) {
+                return void resolve();
+              }
             }
-          }
-          if (--retryCount <= 0) {
-            return void resolve();
-          }
-
+          )
           setTimeout(requestTags, interval);
-        });
+        };
+        requestTags();
```

Comment:
And this is the fourth change: invoking `requestTags` immediately after defining it. Without this, you can see that `requestTags` never gets called in the first place.

These four changes need to be part of any fix. I would also like this formatting change to make it in.

I understand what you're saying about the whitespace. If you're saying that all whitespace-related changes should go into their own commit, that's fine. If you're saying that they shouldn't make it into this PR at all, that'd be harder for me to agree with. But it's not something I'd go twelve rounds over.